### PR TITLE
[lang] MatrixType bug fix: Demote ret_type for ArgLoadStmt after scalarization

### PR DESCRIPTION
**Pull request by jim19930609**
Issue: https://github.com/taichi-dev/taichi/issues/5819

### CODE REVIEW

File: taichi/transforms/scalarize.cpp [hunk 1]
```diff
@@ -47,7 +47,12 @@ class Scalarize : public BasicStmtVisitor {
       auto dest_tensor_type = dest_dtype->template as<TensorType>();
       auto val_tensor_type = val_dtype->template as<TensorType>();
 
-      TI_ASSERT(dest_tensor_type->get_shape() == val_tensor_type->get_shape());
+      // Taichi used to treat VectorType(n) as MatrixType(n, 1),
```
Comment (AD1024) [inline taichi/transforms/scalarize.cpp:50]: Probably worth discussing how we represent Vectors in CHI IR. Would be better to make the semantics consistent across different levels of IRs.

Comment (jim19930609) [reply taichi/transforms/scalarize.cpp:50]: +1 for further discussion over Vector v.s. 1-dimensional Matrix semantics.

The other thing is that `type_check` probably does not allow `Vector(n) op Matrix(n, 1)`, but `scalarization` pass stays synchronized with the legacy `scalarization` happened in Python, which is why we decided to let `Vector(n) op Matrix(n, 1)` pass.

Let's have a thorough discussion then decide what we should do with this PR.

**Review by AD1024 — APPROVED** [review]
LGTM. Thx!

File: taichi/transforms/scalarize.cpp (hunk 1 above)
Comment (AD1024) [reply taichi/transforms/scalarize.cpp:50]: SGTM :)

File: taichi/transforms/lower_matrix_ptr.cpp [hunk 2]
```diff
@@ -97,6 +97,11 @@ class RemoveMatrixOfPtr : public BasicStmtVisitor {
     modifier_.erase(stmt);
   }
 
+  void visit(ArgLoadStmt *stmt) override {
```
Comment (strongoier) [inline taichi/transforms/lower_matrix_ptr.cpp:100]: Hmm.. should we move it elsewhere? `RemoveMatrixOfPtr` is for cleaning redundant stmts after `LowerMatrixPtr`.

**Review by strongoier — APPROVED** [review]
LGTM!

**Pull request merged**